client/components/date-control/index.tsx

```diff
@@ -19,12 +20,13 @@ const DateControl = ( {
 	overlay,
 }: DateControlProps ) => {
 	const moment = useLocalizedMoment();
+	const siteToday = useMomentSiteZone();
 
 	const getShortcutForRange = () => {
 		// Search the shortcut array for something matching the current date range.
 		// Returns shortcut or null;
-		const today = moment().format( 'YYYY-MM-DD' );
-		const yesterday = moment().subtract( 1, 'days' ).format( 'YYYY-MM-DD' );
+		const today = siteToday.format( 'YYYY-MM-DD' );
+		const yesterday = siteToday.clone().subtract( 1, 'days' ).format( 'YYYY-MM-DD' );
```

Comment:
Always clone for mutations ❗❗

Reply:
Yes, good call 👍  Keeps the original data intact and avoids unintended side effects. 🔄